### PR TITLE
Adds Omega alert.

### PR DESCRIPTION
**Pull request by AnCopper**
## About The Pull Request

Adds the Omega alert level.

Partly ports this: https://github.com/yogstation13/Yogstation/pull/9244
Some text loosely inspired from this: https://github.com/yogstation13/Yogstation/pull/12177

It also makes the alert code be a bit more readable.

## Why It's Good For The Game

This is intended to be used as an RP mechanic by admins when they plan to send in a Death Squad or anything else close to that.

It doesn't actually do anything, but I think it's good for flavor and giving you the "Oh fuck" feeling when you hear the distinct siren.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://streamable.com/4krft9

image

</details>

## Changelog
:cl: swissloaf AnCopper
add: Added omega alert.
/:cl:

### CODE REVIEW

**Comment by Kapu1178** [comment]
didnt someone already try to do this and got told "This is just delta alert"

**Comment by Mat05usz** [comment]
> didnt someone already try to do this and got told "This is just delta alert"

I think so, and they were correct. Delta alert is basically 'fuck, we're done'.

Edit: Deathsquad is so rarely called I don't think it even matters. We rarely get CC inspectors, or even low alert ERT, so that's just a cool, but pointless addition imo.

Edit2: 
#5215 
I think this is the PR with alerts from Yog. Omega makes more sense than those but still kinda eeh.

**Comment by Fronsis** [comment]
> > didnt someone already try to do this and got told "This is just delta alert"
> 
> I think so, and they were correct. Delta alert is basically 'fuck, we're done'.
> 
> Edit: Deathsquad is so rarely called I don't think it even matters. We rarely get CC inspectors, or even low alert ERT, so that's just a cool, but pointless addition imo.
> 
> Edit2: #5215 I think this is the PR with alerts from Yog. Omega makes more sense than those but still kinda eeh.

Those are also nice! I'm personally a fan of the ones that you can declare for medical and engineer emergencies, a couple of tg downstreams have it, i don't remember if there's any others besides those (and standards green blue red delta)

**Comment by zeskorion** [comment]
just use delta

**Comment by DatBoiTim** [comment]
I still think it's worth having other non-doomsday alert levels.

Think like Bay's Orange alert for engineering emergencies, ~~purple~~violet alert for biological(xenobio/viro/mass casualty), could probably add one for science specifically.

But aside from that doomsday alert levels are moot.

If you really want to provide admin only alert levels, make a system where they can name the alert level on their own and give it meaning.

**Comment by Daridine** [comment]
> If you really want to provide admin only alert levels, make a system where they can name the alert level on their own and give it meaning.

I would LOVE this.

File: code/modules/security_levels/security_levels.dm [hunk 1]
```diff
@@ -14,71 +15,72 @@ GLOBAL_VAR_INIT(security_level, SEC_LEVEL_GREEN)
...
+						modTimer = 0.5
+
+		GLOB.security_level = level
+		for(var/obj/machinery/firealarm/FA in GLOB.machines)
+			if(is_station_level(FA.z))
+				FA.update_icon()
+
 		if(level >= SEC_LEVEL_RED)
+			for(var/obj/machinery/computer/shuttle_flight/pod/pod in GLOB.machines)
+				pod.admin_controlled = FALSE
```
Comment (Archanial) [inline code/modules/security_levels/security_levels.dm:75]: ?

File: code/controllers/configuration/entries/game_options.dm [hunk 2]
```diff
@@ -235,6 +235,9 @@
 /datum/config_entry/string/alert_red_downto
 	config_entry_value = "The station's destruction has been averted. There is still however an immediate serious threat to the station. Security may have weapons unholstered at all times, random searches are allowed and advised."
 
+/datum/config_entry/string/alert_omega
+	config_entry_value = "Attention, Code Omega has been initiated. Consider all employment contracts terminated, effective immediately."
```
Comment (Archanial) [inline code/controllers/configuration/entries/game_options.dm:239]: We don't need to bloat the config even more.
Just make it static string

**Review by Archanial — CHANGES_REQUESTED** [review]
code looks mostly good at firsts glance. I'll ask admins if they want this feature or not.

**Comment by ItsTisme** [comment]
If possible, make the alert message settable by admins on being called, some extra configuration stuff on runtime could make this from being delta+ to a wild card alert level for either adminbus or reflecting cool thematics of the station

File: code/controllers/configuration/entries/game_options.dm (hunk 2 above)
Comment (Archanial) [reply code/controllers/configuration/entries/game_options.dm:239]: Actually make this alert message be able to be set by admin upon trying to change the alert to this and keep this as default value or wharever

File: code/modules/security_levels/security_levels.dm (hunk 1 above)
Comment (PowerfulBacon) [reply code/modules/security_levels/security_levels.dm:75]: Allows escape pods to be flown

**Comment by PowerfulBacon** [comment]
I dislike this, since it has the same effect as delta and bloats the alert code. If more alerts are going to be added, we need a better alert system with something like a /datum/alert_level rather than if(1), if(2) etc. etc.

**Pull request closed without merging**